search_params.py

```diff
@@ -54,26 +54,19 @@ def init_model_config(config_path):
     return model_config
 
 
-def init_search_space(search_alg, search_func, values):
-    """Initialize the search space by the given search algorithm.
-    Currently, the search algorithm decides what search function is used for all parameters.
+def init_sample_spaces(model_config):
```

Comment:
I think this subroutine should be called init_search_params_spaces because essentially what we did is to get the search space of each parameter

Reply:
Sounds great!